### PR TITLE
Features/ilu 242 UI validation

### PR DESCRIPTION
**Pull request by thatisdrtruman**
This should be pulled now

### CODE REVIEW

File: ips/services/new_run_steps/step_5.py [hunk 1]
```diff
@@ -1,81 +1,104 @@
...
+from flask_wtf.file import FileRequired
 
 from ips.services import app_methods
 from ips.services.forms import LoadDataForm
 
-
 def run_step_5(run_id):
-    form = LoadDataForm()
+    from remote_pdb import RemotePdb
+    #RemotePdb('0.0.0.0', 4445).set_trace()
```
Comment (bitmonkey) [inline ips/services/new_run_steps/step_5.py]: You still seem to have lots of commented out pdb's in your code and imports of remote_pdb, looking for matches with ag inside of the ips_user_interface directory I get:

```
$ ag --python -i pdb
ips/services/new_run_steps/step_5.py
9:    from remote_pdb import RemotePdb
10:    #RemotePdb('0.0.0.0', 4445).set_trace() 
14:    #RemotePdb('0.0.0.0', 4445).set_trace() 
27:    #RemotePdb('0.0.0.0', 4445).set_trace() 

tests/test_step_5.py
23:from remote_pdb import RemotePdb
113:        #RemotePdb('0.0.0.0', 4445).set_trace()
```

File: ips/services/new_run_steps/step_5.py [hunk 2]
```diff
@@ -1,81 +1,104 @@
...
+        'non_response_file': run_info['NR_FILE'],
+        'unsampled_file': run_info['UNSAMPLED_FILE'],
+        'tunnel_file': run_info['TUNNEL_FILE'],
+        'sea_file': run_info['SEA_FILE'],
+        'air_file': run_info['AIR_FILE']
+    }
+    form = LoadDataForm(meta=meta)
+    #RemotePdb('0.0.0.0', 4445).set_trace() 
+
+    if form.validate():
```
Comment (bitmonkey) [inline ips/services/new_run_steps/step_5.py:25]: I don't think this is right? Not sure why you've changed from `validate_on_submit()` to `validate()` - let's discuss.

**Review by bitmonkey — CHANGES_REQUESTED** [review]

File: ips/services/forms.py [hunk 3]
```diff
@@ -116,15 +117,36 @@ class ImportPVForm(FlaskForm):
...
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        fieldnames = [
+            'survey_file', 'shift_file', 'non_response_file', 'unsampled_file',
+            'tunnel_file', 'sea_file', 'air_file' 
+        ]
+    
+        validators = {}
```
Comment (bitmonkey) [inline ips/services/forms.py]: Would be useful to leave a comment here explaining what is happening i.e that you're looping through the fields adding a FileRequired validator to the _extra_validator attribute for fields that haven't had a filename set in the meta object.

**Review by bitmonkey — CHANGES_REQUESTED** [review]

**Review by bitmonkey — APPROVED** [review]

**Comment by thatisdrtruman** [comment]
All good now.

**Pull request merged**